### ISSUE TITLE
switching quickly `tabBarVisible` freezes the app

### ISSUE THREAD

**Issue by ocarreterom** [react-navigation/react-navigation-material-bottom-tabs#31]
The app is frozen when the attribute `tabBarVisible` change quickly in `navigationOptions`.
```js
const routes = {
  MyTab: {
    screen: MyScreen,
    navigationOptions: ({ navigation }) => ({
      tabBarVisible: navigation.state.index === 0,
      tabBarLabel: translate('screen.calendar.title'),
      tabBarIcon: ({ tintColor }) => <Icon name='event' color={tintColor} />
    })
  }
}

createMaterialBottomTabNavigator(routes)
```

When I open a modal screen and close it quickly, the app is frozen and the bottom tabbar starts jumping. If I remove the `tabBarVisible` attribute, all works.

```
Node: 11.6.0
react: 16.6.3
react-native: 0.57.8
react-navigation-material-bottom-tabs@1.0.0
react-navigation@3.0.9
```

**Comment by twhitacre** [comment]
I'm seeing the exact same thing. Here versions if this helps anyone. I've been working on it for a few hours now to no avail.

```
expo: ^30.0.0 => 30.0.1
react: 16.3.1 => 16.3.1
react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4
react-navigation: 2.14.2 => 2.14.2
react-navigation-material-bottom-tabs: 0.4.0
```

I can also verify that turning off `tabBarVisible` will fix the problem, though it's not a good solution because then I can't control which pages show the tabs and which don't.

**Comment by adamawang** [comment]
Happening on my end as well. Seems like it's getting into some sort of resizing/safe-padding loop causing the app to lock up

```
"react": "^16.7.0",
"react-native": "0.57.8",
"react-navigation": "^3.0.9",
"react-navigation-material-bottom-tabs": "^1.0.0",
"react-native-paper": "^2.4.0",
```

**Comment by ocarreterom** [comment]
I will check if it was fixed in https://github.com/callstack/react-native-paper/issues/800

**Comment by ocarreterom** [comment]
@twhitacre @adamawang It's fixed updating `react-native-paper@2.9.0+`.
